gha-webhook-listener.py

```diff
@@ -251,8 +253,14 @@ def deploy_tarball(artifact_url: str, target_dir: str) -> None:
                 raise
 
     logger.info("...download complete.")
+    if arg_symlink:
+        create_symlink(source=target_dir, linkname=arg_symlink)
 
-    create_symlink(source=target_dir, linkname=arg_symlink)
+    if arg_hook_script is not None:
+        logger.info("...running hook script")
+        return_code = subprocess.run([*arg_hook_script.split(), target_dir]).returncode
+        if return_code != 0:
+            logger.info("got return code %i", return_code)
```

Comment:
Thought: should we run the hook script before the symlink step, and if the hook script fails skip the symlinking? Then it's possible to use the hook script as a sort of preparation step before we finally symlink the output into place.